packages/nmap/build.sh

```diff
@@ -2,11 +2,12 @@ TERMUX_PKG_HOMEPAGE=https://nmap.org/
 TERMUX_PKG_DESCRIPTION="Utility for network discovery and security auditing"
 TERMUX_PKG_LICENSE="GPL-2.0"
 TERMUX_PKG_VERSION=7.80
+TERMUX_PKG_REVISION=1
 TERMUX_PKG_SRCURL=https://nmap.org/dist/nmap-${TERMUX_PKG_VERSION}.tar.bz2
 TERMUX_PKG_SHA256=fcfa5a0e42099e12e4bf7a68ebe6fde05553383a682e816a7ec9256ab4773faa
 # Depend on netcat so that it gets installed automatically when installing
 # nmap, since the ncat program is usually distributed as part of nmap.
-TERMUX_PKG_DEPENDS="libc++, libpcap, pcre, openssl, resolv-conf, netcat, liblua, libssh2, zlib"
+TERMUX_PKG_DEPENDS="libc++, libpcap, pcre, openssl, resolv-conf, netcat, liblua53, libssh2, zlib"
```

Comment:
Nmap tries linking with liblua.so which is 5.4:
```
aarch64-linux-android-clang++ -L/data/data/com.termux/files/usr/lib -Wl,-E -L/data/data/com.termux/files/usr/lib -Wl,-rpath=/data/data/com.termux/files/usr/lib -Wl,--enable-new-dtags -Wl,--as-needed -Wl,-z,relro,-z,now -Lnbase -Lnsock/src/   -o nmap charpool.o FingerPrintResults.o FPEngine.o FPModel.o idle_scan.o MACLookup.o nmap_dns.o nmap_error.o nmap.o nmap_ftp.o NmapOps.o NmapOutputTable.o nmap_tty.o osscan2.o osscan.o output.o payload.o portlist.o portreasons.o protocols.o scan_engine.o scan_engine_connect.o scan_engine_raw.o scan_lists.o service_scan.o services.o NewTargets.o TargetGroup.o Target.o targets.o tcpip.o timing.o traceroute.o utils.o xml.o nse_main.o nse_utility.o nse_nsock.o nse_dnet.o nse_fs.o nse_nmaplib.o nse_debug.o nse_pcrelib.o nse_lpeg.o nse_openssl.o nse_ssl_cert.o nse_libssh2.o nse_zlib.o main.o -lnsock -lnbase -lpcre -lpcap -lssh2 -lssl -lcrypto -lz libnetutil/libnetutil.a ./libdnet-stripped/src/.libs/libdnet.a  ./liblinear/liblinear.a -llua 
make[1]: Entering directory '/home/builder/.termux-build/nmap/src/nping'
aarch64-linux-android-clang++ -MM -I../libdnet-stripped/include -I/data/data/com.termux/files/usr/include -I../nbase -I../nsock/include ArgParser.cc common.cc common_modified.cc nping.cc NpingOps.cc utils.cc utils_net.cc output.cc stats.cc NpingTargets.cc NpingTarget.cc EchoHeader.cc EchoServer.cc EchoClient.cc ProbeMode.cc NEPContext.cc Crypto.cc > makefile.dep
nse_nsock.o:nse_nsock.cc:function l_pcap_open(lua_State*): error: undefined reference to 'lua_getuservalue'
nse_nsock.o:nse_nsock.cc:function l_sleep(lua_State*): error: undefined reference to 'lua_newuserdata'
nse_nsock.o:nse_nsock.cc:function initialize(lua_State*, int, nse_nsock_udata*, int, int): error: undefined reference to 'lua_setuservalue'
nse_nsock.o:nse_nsock.cc:function yield(lua_State*, nse_nsock_udata*, char const*, char const*, int, int (*)(lua_State*, int, long)): error: undefined reference to 'lua_getuservalue'
nse_nsock.o:nse_nsock.cc:function receive_buf(lua_State*, int, long): error: undefined reference to 'lua_getuservalue'
nse_nsock.o:nse_nsock.cc:function OUTLINED_FUNCTION_27: error: undefined reference to 'lua_newuserdata'
nse_dnet.o:nse_dnet.cc:function l_dnet_new(lua_State*): error: undefined reference to 'lua_newuserdata'
nse_dnet.o:nse_dnet.cc:function open_eth_cached(lua_State*, int, char const*): error: undefined reference to 'lua_newuserdata'
nse_lpeg.o:nse_lpeg.cc:function getpatt(lua_State*, int, int*): error: undefined reference to 'lua_setuservalue'
nse_lpeg.o:nse_lpeg.cc:function getpatt(lua_State*, int, int*): error: undefined reference to 'lua_getuservalue'
nse_lpeg.o:nse_lpeg.cc:function copyktable(lua_State*, int): error: undefined reference to 'lua_setuservalue'
nse_lpeg.o:nse_lpeg.cc:function OUTLINED_FUNCTION_45: error: undefined reference to 'lua_setuservalue'
```
I guess NSE modules wouldn't like Lua 5.4 too.

Reply:
Fixed.